File: functions-python/astro/vedic_constants.py
"""
Vedic Astrology Constants and Interpretation Library
Complete Jyotish interpretation tables for Lagna, Nakshatra, Graha, Bhava, and Dasha
"""

# ============================================================================
# LAGNA (ASCENDANT) INTERPRETATIONS
# ============================================================================

LAGNA_INTERPRETATIONS = {
    "Mesha": "An Aries (Mesha) Lagna brings a direct, pioneering, and action-oriented approach to life. You tend to initiate quickly and learn through experience. Mars rules your chart, giving courage and competitive spirit.",
    "Vrishabha": "A Taurus (Vrishabha) Lagna values stability, comfort, and tangible results. You move steadily and prefer to build lasting foundations. Venus rules your chart, emphasizing beauty and material security.",
    "Mithuna": "A Gemini (Mithuna) Lagna is curious, communicative, and mentally agile. You learn through interaction and variety. Mercury rules your chart, giving intellectual adaptability.",
    "Karka": "A Cancer (Karka) Lagna is sensitive, protective, and nurturing. Emotional security and home are central themes. The Moon rules your chart, emphasizing emotional intelligence.",
    "Simha": "A Leo (Simha) Lagna is expressive, proud, and creative. You naturally seek to shine and lead from the heart. The Sun rules your chart, giving authority and vitality.",
    "Kanya": "A Virgo (Kanya) Lagna is analytical, precise, and service-oriented. You refine, improve, and pay attention to details. Mercury rules your chart with a practical focus.",
    "Tula": "A Libra (Tula) Lagna is relational, diplomatic, and balance-seeking. Partnerships and harmony are key life themes. Venus rules your chart, emphasizing relationships.",
    "Vrishchika": "A Scorpio (Vrishchika) Lagna is intense, private, and transformative. You move through deep emotional cycles and regeneration. Mars rules your chart with depth and power.",
    "Dhanu": "A Sagittarius (Dhanu) Lagna is philosophical, adventurous, and future-oriented. You seek meaning, learning, and expansion. Jupiter rules your chart, giving wisdom and optimism.",
    "Makara": "A Capricorn (Makara) Lagna is disciplined, pragmatic, and status-aware. You build slowly toward long-term goals. Saturn rules your chart, emphasizing responsibility.",
    "Kumbha": "An Aquarius (Kumbha) Lagna is unconventional, idealistic, and community-minded. You think ahead of your time. Saturn rules your chart with a humanitarian focus.",
    "Meena": "A Pisces (Meena) Lagna is imaginative, sensitive, and spiritual. You navigate life through intuition and compassion. Jupiter rules your chart, giving spiritual depth.",
}

# ============================================================================
# NAKSHATRA INTERPRETATIONS
# ============================================================================

NAKSHATRA_INTERPRETATIONS = {
    "Ashwini": "Ashwini brings speed, initiative, and healing energy. Emotionally, there is a drive to start things and move quickly. The Ashwini Kumaras bestow the power to heal and renew.",
    "Bharani": "Bharani carries themes of responsibility, intensity, and transformation. Emotional life can feel deep and consequential. Yama, the lord of dharma, governs this star.",
    "Krittika": "Krittika is sharp, discerning, and protective. There is a strong will to cut through confusion and refine. Agni's fire purifies and illuminates.",
    "Rohini": "Rohini is fertile, attractive, and growth-oriented. Emotional nature seeks beauty, comfort, and creative expression. Brahma's creativity flows through this nakshatra.",
    "Mrigashira": "Mrigashira is searching, curious, and restless. The mind explores, questions, and seeks new experiences. Soma brings a gentle, questing nature.",
    "Ardra": "Ardra is intense, stormy, and transformative. Emotional life may pass through catharsis and deep change. Rudra's power brings destruction that clears the path for renewal.",
    "Punarvasu": "Punarvasu is renewing, hopeful, and resilient. There is a capacity to recover and begin again. Aditi, the cosmic mother, offers restoration.",
    "Pushya": "Pushya is nurturing, supportive, and devotional. Emotional fulfillment comes through caring and guidance. This is considered the most auspicious nakshatra.",
    "Ashlesha": "Ashlesha is penetrating, binding, and psychologically deep. Emotions can be complex and intense. The serpent energy gives insight into hidden realms.",
    "Magha": "Magha is regal, ancestral, and status-conscious. Emotional life is tied to legacy and recognition. The Pitris (ancestors) bestow authority and dignity.",
    "Purva Phalguni": "Purva Phalguni is pleasure-seeking, creative, and relational. There is a love of enjoyment and connection. Bhaga brings good fortune in relationships.",
    "Uttara Phalguni": "Uttara Phalguni is generous, responsible, and stabilizing. Relationships and agreements are central. Aryaman governs contracts and friendship.",
    "Hasta": "Hasta is skillful, clever, and hands-on. Emotional nature expresses through doing, crafting, and managing. Savitar brings skill and dexterity.",
    "Chitra": "Chitra is artistic, striking, and individualistic. There is a drive to create and stand out. Vishvakarman, the celestial architect, gives creative vision.",
    "Swati": "Swati is independent, flexible, and wind-like. Emotional life values freedom and self-direction. Vayu brings adaptability and movement.",
    "Vishakha": "Vishakha is goal-driven, determined, and focused. There is intensity in pursuing chosen aims. Indra and Agni together give conquering power.",
    "Anuradha": "Anuradha is loyal, devotional, and friendship-oriented. Emotional fulfillment comes through bonds and shared purpose. Mitra, the god of friendship, governs here.",
    "Jyeshtha": "Jyeshtha is protective, senior, and responsible. Emotions may carry themes of duty and authority. Indra bestows leadership and protective power.",
    "Mula": "Mula is root-seeking, radical, and transformative. Emotional life may involve deep uprooting and truth-seeking. Nirriti brings the power to destroy and regenerate.",
    "Purva Ashadha": "Purva Ashadha is victorious, idealistic, and persuasive. There is a drive to champion beliefs. Apas, the water deity, brings purification and invincibility.",
    "Uttara Ashadha": "Uttara Ashadha is enduring, principled, and leadership-oriented. Emotional nature seeks lasting achievement. The Vishvadevas bestow universal qualities.",
    "Shravana": "Shravana is listening, learning, and reputation-focused. Emotional fulfillment comes through knowledge and recognition. Vishnu brings the power of connection.",
    "Dhanishta": "Dhanishta is rhythmic, communal, and achievement-oriented. There is a drive to perform and contribute. The Vasus bring abundance and fame.",
    "Shatabhisha": "Shatabhisha is solitary, healing, and boundary-focused. Emotional life may involve withdrawal and deep repair. Varuna governs the cosmic waters.",
    "Purva Bhadrapada": "Purva Bhadrapada is intense, idealistic, and transformative. There can be extremes in belief and emotion. Aja Ekapada brings fierce one-pointed energy.",
    "Uttara Bhadrapada": "Uttara Bhadrapada is stabilizing, deep, and contemplative. Emotional nature leans toward inner work. Ahir Budhnya brings wisdom from the depths.",
    "Revati": "Revati is gentle, protective, and guiding. Emotional fulfillment comes through support, travel, and completion. Pushan guides safe journeys and nurtures growth.",
}

# ============================================================================
# GRAHA (PLANET) BASE THEMES
# ============================================================================

GRAHA_BASE_THEMES = {
    "surya": "The Sun represents identity, vitality, authority, father, and soul purpose. It shows your core sense of self and how you seek recognition.",
    "chandra": "The Moon represents mind, emotions, mother, habits, and inner security. It shows your emotional nature and how you nurture yourself and others.",
    "mangala": "Mars represents drive, courage, conflict, siblings, and technical skill. It shows how you assert yourself and pursue goals with energy.",
    "budha": "Mercury represents intellect, communication, trade, and adaptability. It shows how you think, learn, and express ideas.",
    "guru": "Jupiter represents wisdom, growth, faith, teachers, and blessings. It shows where expansion, optimism, and good fortune flow in your life.",
    "shukra": "Venus represents love, beauty, pleasure, art, and harmony. It shows your capacity for relationships, creativity, and material enjoyment.",
    "shani": "Saturn represents discipline, structure, karma, delays, and endurance. It shows where you face tests and build lasting foundations.",
    "rahu": "Rahu represents obsession, innovation, foreignness, and worldly desires. It shows where you break conventions and pursue ambitious goals.",
    "ketu": "Ketu represents detachment, spirituality, past-life residue, and sudden breaks. It shows where you release attachments and seek liberation.",
}

# ============================================================================
# BHAVA (HOUSE) MEANINGS
# ============================================================================

BHAVA_MEANINGS = {
    1: "1st Bhava (Tanu): Self, body, identity, health, and overall life direction. The foundation of your entire chart.",
    2: "2nd Bhava (Dhana): Wealth, speech, family values, food, and accumulated resources. Your capacity to sustain yourself.",
    3: "3rd Bhava (Sahaja): Courage, siblings, communication, effort, and short journeys. Your will and initiative.",
    4: "4th Bhava (Sukha): Home, mother, emotional foundation, property, and inner peace. Your roots and comfort.",
    5: "5th Bhava (Putra): Creativity, children, intelligence, romance, and past-life merit. Your creative expression.",
    6: "6th Bhava (Ari): Work, service, health issues, obstacles, enemies, and daily routines. Where you overcome challenges.",
    7: "7th Bhava (Yuvati): Partnerships, marriage, contracts, business, and public dealings. Your one-to-one relationships.",
    8: "8th Bhava (Randhra): Transformation, secrets, longevity, inheritance, and shared resources. Deep change and hidden matters.",
    9: "9th Bhava (Dharma): Higher learning, teachers, father, fortune, philosophy, and long journeys. Your path of wisdom.",
    10: "10th Bhava (Karma): Career, status, reputation, authority, and public role. Your visible achievements and purpose.",
    11: "11th Bhava (Labha): Gains, networks, aspirations, elder siblings, and friendships. Where you receive and connect.",
    12: "12th Bhava (Vyaya): Loss, liberation, foreign lands, expenses, and spiritual retreat. Where you release and transcend.",
}

# ============================================================================
# DASHA PLANET THEMES
# ============================================================================

DASHA_PLANET_THEMES = {
    "surya": "Sun Mahadasha (6 years) emphasizes identity, leadership, visibility, father, government, and questions of purpose and authority. A time to shine and claim your place.",
    "chandra": "Moon Mahadasha (10 years) emphasizes emotional life, home, mother, family, public image, and inner security. A time of fluctuation and emotional growth.",
    "mangala": "Mars Mahadasha (7 years) emphasizes action, conflict, courage, property, siblings, and technical or physical pursuits. A time of energy and assertion.",
    "budha": "Mercury Mahadasha (17 years) emphasizes learning, communication, trade, intellect, and adaptability. A long period of mental development and commerce.",
    "guru": "Jupiter Mahadasha (16 years) emphasizes growth, wisdom, children, teaching, and expansion of fortune. A blessed period of opportunity and dharma.",
    "shukra": "Venus Mahadasha (20 years) emphasizes relationships, pleasure, art, wealth, and material comforts. The longest dasha, focused on enjoyment and connection.",
    "shani": "Saturn Mahadasha (19 years) emphasizes responsibility, discipline, delays, karma, and long-term consolidation. A period of testing and maturation.",
    "rahu": "Rahu Mahadasha (18 years) emphasizes worldly ambition, unconventional paths, foreign connections, and intense desires. A time of rapid, often unexpected change.",
    "ketu": "Ketu Mahadasha (7 years) emphasizes detachment, spiritualization, sudden events, and completion of old karmas. A period of release and inner focus.",
}

# ============================================================================
# GUNA (MENTAL QUALITY) CONSTANTS
# ============================================================================

GUNA_BY_SIGN = {
    "Mesha": "Sattva", "Simha": "Sattva", "Dhanu": "Sattva",
    "Vrishabha": "Rajas", "Kanya": "Rajas", "Makara": "Rajas",
    "Mithuna": "Rajas", "Tula": "Rajas", "Kumbha": "Rajas",
    "Karka": "Tamas", "Vrishchika": "Tamas", "Meena": "Tamas",
}

GUNA_INTERPRETATIONS = {
    "Sattva": "Sattvic temperament brings clarity, harmony, idealism, and a naturally balanced mind. You tend toward purity, truth-seeking, and spiritual orientation.",
    "Rajas": "Rajasic temperament brings activity, ambition, desire, and a dynamic, restless mind. You are driven, creative, and oriented toward achievement and action.",
    "Tamas": "Tamasic temperament brings depth, intensity, introspection, and emotional complexity. You process life deeply and may need more time for rest and reflection.",
}

# ============================================================================
# DOSHA (AYURVEDIC CONSTITUTION) CONSTANTS
# ============================================================================

DOSHA_BY_SIGN = {
    "Mesha": "Pitta", "Simha": "Pitta", "Dhanu": "Pitta", "Vrishchika": "Pitta",
    "Mithuna": "Vata", "Tula": "Vata", "Kumbha": "Vata",
    "Vrishabha": "Kapha", "Kanya": "Kapha", "Makara": "Kapha",
    "Karka": "Kapha", "Meena": "Kapha",
}

DOSHA_INTERPRETATIONS = {
    "Vata": "Vata constitution brings creativity, quick thinking, intuition, and movement. You may be prone to anxiety, irregularity, or scattered energy when imbalanced.",
    "Pitta": "Pitta constitution brings focus, intensity, sharp intellect, and strong willpower. You may be prone to anger, criticism, or burnout when imbalanced.",
    "Kapha": "Kapha constitution brings steadiness, nurturing energy, calm, and emotional resilience. You may be prone to lethargy, attachment, or resistance to change when imbalanced.",
}

# ============================================================================
# GRAHA DIGNITY (OWN SIGN, EXALTATION, DEBILITATION)
# ============================================================================

GRAHA_OWN_SIGNS = {
    "surya": ["Simha"],
    "chandra": ["Karka"],
    "mangala": ["Mesha", "Vrishchika"],
    "budha": ["Mithuna", "Kanya"],
    "guru": ["Dhanu", "Meena"],
    "shukra": ["Vrishabha", "Tula"],
    "shani": ["Makara", "Kumbha"],
    "rahu": [],
    "ketu": [],
}

GRAHA_EXALTATION = {
    "surya": "Mesha",
    "chandra": "Vrishabha",
    "mangala": "Makara",
    "budha": "Kanya",
    "guru": "Karka",
    "shukra": "Meena",
    "shani": "Tula",
    "rahu": "Vrishabha",
    "ketu": "Vrishchika",
}

GRAHA_DEBILITATION = {
    "surya": "Tula",
    "chandra": "Vrishchika",
    "mangala": "Karka",
    "budha": "Meena",
    "guru": "Makara",
    "shukra": "Kanya",
    "shani": "Mesha",
    "rahu": "Vrishchika",
    "ketu": "Vrishabha",
}

File: src/data/vedicConstants.js
/**
 * Vedic Astrology (Jyotish) Constants
 * Used for sidereal zodiac calculations with Swiss Ephemeris
 */

// ============================================================================
// RASHIS (Zodiac Signs in Vedic Astrology)
// ============================================================================

export const RASHIS = [
  { index: 0, sanskrit: 'Mesha', english: 'Aries', lord: 'Mars', element: 'Fire', symbol: '♈', emoji: '🐏', quality: 'Movable' },
  { index: 1, sanskrit: 'Vrishabha', english: 'Taurus', lord: 'Venus', element: 'Earth', symbol: '♉', emoji: '🐂', quality: 'Fixed' },
  { index: 2, sanskrit: 'Mithuna', english: 'Gemini', lord: 'Mercury', element: 'Air', symbol: '♊', emoji: '👯', quality: 'Dual' },
  { index: 3, sanskrit: 'Karka', english: 'Cancer', lord: 'Moon', element: 'Water', symbol: '♋', emoji: '🦀', quality: 'Movable' },
  { index: 4, sanskrit: 'Simha', english: 'Leo', lord: 'Sun', element: 'Fire', symbol: '♌', emoji: '🦁', quality: 'Fixed' },
  { index: 5, sanskrit: 'Kanya', english: 'Virgo', lord: 'Mercury', element: 'Earth', symbol: '♍', emoji: '👸', quality: 'Dual' },
  { index: 6, sanskrit: 'Tula', english: 'Libra', lord: 'Venus', element: 'Air', symbol: '♎', emoji: '⚖️', quality: 'Movable' },
  { index: 7, sanskrit: 'Vrishchika', english: 'Scorpio', lord: 'Mars', element: 'Water', symbol: '♏', emoji: '🦂', quality: 'Fixed' },
  { index: 8, sanskrit: 'Dhanu', english: 'Sagittarius', lord: 'Jupiter', element: 'Fire', symbol: '♐', emoji: '🏹', quality: 'Dual' },
  { index: 9, sanskrit: 'Makara', english: 'Capricorn', lord: 'Saturn', element: 'Earth', symbol: '♑', emoji: '🐐', quality: 'Movable' },
  { index: 10, sanskrit: 'Kumbha', english: 'Aquarius', lord: 'Saturn', element: 'Air', symbol: '♒', emoji: '🏺', quality: 'Fixed' },
  { index: 11, sanskrit: 'Meena', english: 'Pisces', lord: 'Jupiter', element: 'Water', symbol: '♓', emoji: '🐟', quality: 'Dual' },
];

// ============================================================================
// NAKSHATRAS (27 Lunar Mansions)
// Each spans 13°20' (13.333...°), with 4 padas of 3°20' each
// ============================================================================

export const NAKSHATRAS = [
  { index: 0, name: 'Ashwini', lord: 'Ketu', deity: 'Ashwini Kumaras', symbol: '🐴', quality: 'Light/Swift',
    meaning: 'The Horse-headed Twins', nature: 'Healing, speed, new beginnings' },
  { index: 1, name: 'Bharani', lord: 'Venus', deity: 'Yama', symbol: '🔺', quality: 'Fierce/Severe',
    meaning: 'The Bearer', nature: 'Transformation, restraint, duty' },
  { index: 2, name: 'Krittika', lord: 'Sun', deity: 'Agni', symbol: '🔥', quality: 'Mixed',
    meaning: 'The Cutter', nature: 'Purification, determination, courage' },
  { index: 3, name: 'Rohini', lord: 'Moon', deity: 'Brahma', symbol: '🐂', quality: 'Fixed/Permanent',
    meaning: 'The Red One', nature: 'Creativity, fertility, growth' },
  { index: 4, name: 'Mrigashira', lord: 'Mars', deity: 'Soma', symbol: '🦌', quality: 'Soft/Mild',
    meaning: 'The Deer\'s Head', nature: 'Searching, curiosity, gentleness' },
  { index: 5, name: 'Ardra', lord: 'Rahu', deity: 'Rudra', symbol: '💎', quality: 'Sharp/Dreadful',
    meaning: 'The Moist One', nature: 'Storms, destruction, renewal' },
  { index: 6, name: 'Punarvasu', lord: 'Jupiter', deity: 'Aditi', symbol: '🏹', quality: 'Movable/Ephemeral',
    meaning: 'Return of the Light', nature: 'Restoration, abundance, renewal' },
  { index: 7, name: 'Pushya', lord: 'Saturn', deity: 'Brihaspati', symbol: '🌸', quality: 'Light/Swift',
    meaning: 'The Nourisher', nature: 'Nourishment, protection, prosperity' },
  { index: 8, name: 'Ashlesha', lord: 'Mercury', deity: 'Sarpas', symbol: '🐍', quality: 'Sharp/Dreadful',
    meaning: 'The Embracer', nature: 'Kundalini, intuition, mysticism' },
  { index: 9, name: 'Magha', lord: 'Ketu', deity: 'Pitris', symbol: '👑', quality: 'Fierce/Severe',
    meaning: 'The Great One', nature: 'Royalty, ancestors, authority' },
  { index: 10, name: 'Purva Phalguni', lord: 'Venus', deity: 'Bhaga', symbol: '🛏️', quality: 'Fierce/Severe',
    meaning: 'Former Red One', nature: 'Pleasure, creativity, relationships' },
  { index: 11, name: 'Uttara Phalguni', lord: 'Sun', deity: 'Aryaman', symbol: '🛏️', quality: 'Fixed/Permanent',
    meaning: 'Latter Red One', nature: 'Contracts, partnerships, patronage' },
  { index: 12, name: 'Hasta', lord: 'Moon', deity: 'Savitar', symbol: '✋', quality: 'Light/Swift',
    meaning: 'The Hand', nature: 'Skill, dexterity, craftsmanship' },
  { index: 13, name: 'Chitra', lord: 'Mars', deity: 'Vishvakarman', symbol: '💎', quality: 'Soft/Mild',
    meaning: 'The Bright One', nature: 'Artistry, beauty, illusion' },
  { index: 14, name: 'Swati', lord: 'Rahu', deity: 'Vayu', symbol: '🌱', quality: 'Movable/Ephemeral',
    meaning: 'The Sword', nature: 'Independence, flexibility, trade' },
  { index: 15, name: 'Vishakha', lord: 'Jupiter', deity: 'Indra-Agni', symbol: '🎯', quality: 'Mixed',
    meaning: 'The Forked One', nature: 'Goal-oriented, triumph, ambition' },
  { index: 16, name: 'Anuradha', lord: 'Saturn', deity: 'Mitra', symbol: '🪷', quality: 'Soft/Mild',
    meaning: 'Following Radha', nature: 'Devotion, friendship, success' },
  { index: 17, name: 'Jyeshtha', lord: 'Mercury', deity: 'Indra', symbol: '👂', quality: 'Sharp/Dreadful',
    meaning: 'The Eldest', nature: 'Leadership, protection, authority' },
  { index: 18, name: 'Mula', lord: 'Ketu', deity: 'Nirriti', symbol: '🦁', quality: 'Sharp/Dreadful',
    meaning: 'The Root', nature: 'Investigation, dissolution, roots' },
  { index: 19, name: 'Purva Ashadha', lord: 'Venus', deity: 'Apas', symbol: '🐘', quality: 'Fierce/Severe',
    meaning: 'Former Invincible', nature: 'Purification, invincibility, revival' },
  { index: 20, name: 'Uttara Ashadha', lord: 'Sun', deity: 'Vishvadevas', symbol: '🐘', quality: 'Fixed/Permanent',
    meaning: 'Latter Invincible', nature: 'Victory, leadership, permanence' },
  { index: 21, name: 'Shravana', lord: 'Moon', deity: 'Vishnu', symbol: '👂', quality: 'Movable/Ephemeral',
    meaning: 'The Listener', nature: 'Learning, connections, perseverance' },
  { index: 22, name: 'Dhanishta', lord: 'Mars', deity: 'Vasus', symbol: '🥁', quality: 'Movable/Ephemeral',
    meaning: 'The Wealthiest', nature: 'Wealth, music, fame' },
  { index: 23, name: 'Shatabhisha', lord: 'Rahu', deity: 'Varuna', symbol: '⭕', quality: 'Movable/Ephemeral',
    meaning: 'Hundred Healers', nature: 'Healing, seclusion, mysteries' },
  { index: 24, name: 'Purva Bhadrapada', lord: 'Jupiter', deity: 'Aja Ekapada', symbol: '🛏️', quality: 'Fierce/Severe',
    meaning: 'Former Lucky Feet', nature: 'Passion, transformation, fire' },
  { index: 25, name: 'Uttara Bhadrapada', lord: 'Saturn', deity: 'Ahir Budhnya', symbol: '🛏️', quality: 'Fixed/Permanent',
    meaning: 'Latter Lucky Feet', nature: 'Depth, wisdom, restraint' },
  { index: 26, name: 'Revati', lord: 'Mercury', deity: 'Pushan', symbol: '🐟', quality: 'Soft/Mild',
    meaning: 'The Wealthy', nature: 'Safe travel, nourishment, completion' },
];

// ============================================================================
// GRAHAS (Vedic Planets)
// ============================================================================

export const GRAHAS = {
  surya: { sanskrit: 'Surya', english: 'Sun', symbol: '☉', color: '#FFD700', nature: 'Malefic', element: 'Fire' },
  chandra: { sanskrit: 'Chandra', english: 'Moon', symbol: '☽', color: '#C0C0C0', nature: 'Benefic', element: 'Water' },
  mangala: { sanskrit: 'Mangala', english: 'Mars', symbol: '♂', color: '#FF4500', nature: 'Malefic', element: 'Fire' },
  budha: { sanskrit: 'Budha', english: 'Mercury', symbol: '☿', color: '#32CD32', nature: 'Neutral', element: 'Earth' },
  guru: { sanskrit: 'Guru', english: 'Jupiter', symbol: '♃', color: '#FFD700', nature: 'Benefic', element: 'Ether' },
  shukra: { sanskrit: 'Shukra', english: 'Venus', symbol: '♀', color: '#FF69B4', nature: 'Benefic', element: 'Water' },
  shani: { sanskrit: 'Shani', english: 'Saturn', symbol: '♄', color: '#4169E1', nature: 'Malefic', element: 'Air' },
  rahu: { sanskrit: 'Rahu', english: 'North Node', symbol: '☊', color: '#8B4513', nature: 'Malefic', element: 'Air' },
  ketu: { sanskrit: 'Ketu', english: 'South Node', symbol: '☋', color: '#808080', nature: 'Malefic', element: 'Fire' },
};

// ============================================================================
// AYANAMSHA OPTIONS
// ============================================================================

export const AYANAMSHAS = {
  lahiri: { id: 'lahiri', name: 'Lahiri (Chitrapaksha)', description: 'Most widely used, official Indian calendar' },
  raman: { id: 'raman', name: 'B.V. Raman', description: 'Popular in South India' },
  krishnamurti: { id: 'krishnamurti', name: 'Krishnamurti (KP)', description: 'Used in KP Astrology system' },
  fagan_bradley: { id: 'fagan_bradley', name: 'Fagan-Bradley', description: 'Western sidereal astrology' },
  yukteshwar: { id: 'yukteshwar', name: 'Sri Yukteshwar', description: 'Based on Autobiography of a Yogi' },
  true_chitra: { id: 'true_chitra', name: 'True Chitra', description: 'Spica at exactly 0° Libra' },
};

// ============================================================================
// BHAVA (House) MEANINGS
// ============================================================================

export const BHAVAS = {
  1: { name: 'Tanu Bhava', meaning: 'Self & Body', keywords: ['personality', 'appearance', 'health', 'beginnings'] },
  2: { name: 'Dhana Bhava', meaning: 'Wealth & Family', keywords: ['money', 'speech', 'family values', 'food'] },
  3: { name: 'Sahaja Bhava', meaning: 'Siblings & Courage', keywords: ['siblings', 'communication', 'short trips', 'skills'] },
  4: { name: 'Sukha Bhava', meaning: 'Happiness & Mother', keywords: ['home', 'mother', 'emotions', 'property'] },
  5: { name: 'Putra Bhava', meaning: 'Children & Intelligence', keywords: ['children', 'creativity', 'romance', 'education'] },
  6: { name: 'Ari Bhava', meaning: 'Enemies & Health', keywords: ['enemies', 'disease', 'service', 'debts'] },
  7: { name: 'Yuvati Bhava', meaning: 'Spouse & Partnership', keywords: ['marriage', 'business', 'contracts', 'others'] },
  8: { name: 'Randhra Bhava', meaning: 'Longevity & Transformation', keywords: ['death', 'inheritance', 'occult', 'sudden events'] },
  9: { name: 'Dharma Bhava', meaning: 'Fortune & Spirituality', keywords: ['luck', 'father', 'religion', 'higher learning'] },
  10: { name: 'Karma Bhava', meaning: 'Career & Status', keywords: ['profession', 'reputation', 'authority', 'government'] },
  11: { name: 'Labha Bhava', meaning: 'Gains & Friends', keywords: ['income', 'friends', 'hopes', 'elder siblings'] },
  12: { name: 'Vyaya Bhava', meaning: 'Loss & Liberation', keywords: ['expenses', 'foreign', 'spirituality', 'endings'] },
};

// ============================================================================
// ELEMENT COLORS (for UI)
// ============================================================================

export const VEDIC_ELEMENT_COLORS = {
  Fire: { bg: 'bg-orange-500/20', border: 'border-orange-500/40', text: 'text-orange-400' },
  Earth: { bg: 'bg-lime-500/20', border: 'border-lime-500/40', text: 'text-lime-400' },
  Air: { bg: 'bg-sky-500/20', border: 'border-sky-500/40', text: 'text-sky-400' },
  Water: { bg: 'bg-cyan-500/20', border: 'border-cyan-500/40', text: 'text-cyan-400' },
  Ether: { bg: 'bg-purple-500/20', border: 'border-purple-500/40', text: 'text-purple-400' },
};

// ============================================================================
// HELPER FUNCTIONS
// ============================================================================

/**
 * Get Rashi by index (0-11)
 */
export function getRashiByIndex(index) {
  return RASHIS[index % 12];
}

/**
 * Get Nakshatra by index (0-26)
 */
export function getNakshatraByIndex(index) {
  return NAKSHATRAS[index % 27];
}

/**
 * Calculate Nakshatra from sidereal longitude
 */
export function calculateNakshatra(longitude) {
  const nakshatraSpan = 360 / 27; // 13.333...°
  const padaSpan = nakshatraSpan / 4; // 3.333...°

  const index = Math.floor(longitude / nakshatraSpan) % 27;
  const degreeInNakshatra = longitude % nakshatraSpan;
  const pada = Math.floor(degreeInNakshatra / padaSpan) + 1;

  return {
    ...NAKSHATRAS[index],
    pada,
    degreeInNakshatra,
  };
}

/**
 * Get Graha info by key
 */
export function getGrahaInfo(grahaKey) {
  return GRAHAS[grahaKey] || null;
}
